pkg/api/machine.go

```diff
@@ -28,13 +28,16 @@ type MachineSpec struct {
 	NetworkInterfaces []*NetworkInterfaceSpec `json:"networkInterfaces"`
 
 	ShutdownAt time.Time `json:"shutdownAt,omitempty"`
+
+	QemuGuestAgentEnable bool
 }
 
 type MachineStatus struct {
 	VolumeStatus           []VolumeStatus           `json:"volumeStatus"`
 	NetworkInterfaceStatus []NetworkInterfaceStatus `json:"networkInterfaceStatus"`
 	State                  MachineState             `json:"state"`
 	ImageRef               string                   `json:"imageRef"`
+	QemuGuestAgent         *AgentStatus             `json:"qemuGuestAgent,omitempty"`
```

Comment:
In general: Do we need `Qemu` in `QemuGuestAgent` ? Isn't that implementation detail?

Reply:
Theoretically you can have multiple type of agents (https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.3/html/virtual_machine_management_guide/installing_guest_agents_and_drivers_linux#doc-wrapper). But we support qemu currently so i wanted do it extendable for future.

Reply:
but we can do it with extend agentStatus, so yes we can rename it to GuestAgent.

Reply:
I think it's a good idea to put the type of the `GuestAgent` into the status.

Reply:
fixed